quarkus/config-api/src/main/java/org/keycloak/config/WildcardOptionsUtil.java

```diff
@@ -0,0 +1,121 @@
+package org.keycloak.config;
+
+/**
+ * Utility class for working with configuration options that use wildcards.
+ * <p>
+ * Wildcard options are configuration keys that contain a variable segment enclosed between {@link #WILDCARD_START} and {@link #WILDCARD_END} characters.
+ * <p>
+ * Wildcard options in Keycloak always <strong>end</strong> with the variable segment.
+ */
+public class WildcardOptionsUtil {
+
+    /**
+     * Marker indicating the start of a wildcard segment in a configuration key.
+     */
+    public static final String WILDCARD_START = "<";
+
+    /**
+     * Marker indicating the end of a wildcard segment in a configuration key.
+     */
+    public static final String WILDCARD_END = ">";
+
+        /**
+     * Determines whether the given configuration key represents a wildcard option.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * isKcWildcardOption("tracing-header-<header>")      → "true"
+     * isKcWildcardOption("tracing-header-<headxxx")      → "false"
+     * isKcWildcardOption("tracing-header-headxxx>")      → "false"
+     * isKcWildcardOption("db-kind-<datasource>")         → "true"
+     * isKcWildcardOption("http-port")                    → "false"
+     * isKcWildcardOption("quarkus.<sth>.end")            → "false"
```

Comment:
Are you sure we can make this assumption in general?

Reply:
My concern was specifically with "quarkus.<sth>.end" as that seems to be mixing both qualified and unqualified keys - if we expect quarkus., shouldn't all of them something like kc. or smallrye.?

Then are we sure that quarkus options cannot end with a wildcard, and will always have ".end"?